hazelcast/src/main/java/com/hazelcast/scheduledexecutor/StatefulTask.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.scheduledexecutor;
+
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+/**
+ * An interface to provide means for saving & loading state for {@link Runnable} and {@link java.util.concurrent.Callable}
+ * tasks scheduled with an {@link IScheduledExecutorService}. When task implements this interface, the Scheduled Executor
+ * will be able to handle state of the task among the replicas in an event of Migration or Node failure.
+ *
+ * Example:
+ * <pre>
+ * public class CleanUpTask implements Runnable, StatefulTask<String, Integer>, HazelcastInstanceAware {
+ *
+ *      transient HazelcastInstance instance;
```

Comment:
I would make this private to prevent people thinking they need to make it package friendly.